src/main/java/dk/cs/aau/huppaal/logging/Log.java

```diff
@@ -0,0 +1,99 @@
+package dk.cs.aau.huppaal.logging;
+
+import dk.cs.aau.huppaal.BuildConfig;
+import dk.cs.aau.huppaal.utility.FuncInterfaces.Runnable1;
+import javafx.application.Platform;
+
+import java.util.*;
+
+/**
+ * Logging framework for all your messaging needs. A log entry consists of:
+ * <ul>
+ *  <li>{@link #id} an (autogenerated) uuid</li>
+ *  <li>{@link #service} a string of the name of the service that produced the log</li>
+ *  <li>{@link #message} the message to log. See below for formatting</li>
+ *  <li>{@link #level} what severity the log is</li>
+ * </ul>
+ * */
+public record Log(
+        UUID id,
+        String service,         // service that produced the service
+        String message,         // information about what happened
+        LogLevel level          // buffer to put the message into
+) {
+    public static final String DEFAULT_SERVICE = BuildConfig.NAME;
+    public static final Map<String, List<Log>> logs = new HashMap<>();
+    private static final List<Runnable1<Log>> onLogAddedSubscribers = new ArrayList<>();
+    private static final List<Runnable> onLogRemovedSubscribers = new ArrayList<>();
+
+    public static void addError(String message) {
+        addLog(new Log(UUID.randomUUID(), DEFAULT_SERVICE, message, LogLevel.Error));
+    }
+    public static void addWarning(String message) {
+        addLog(new Log(UUID.randomUUID(), DEFAULT_SERVICE, message, LogLevel.Warning));
+    }
+    public static void addInfo(String message) {
+        addLog(new Log(UUID.randomUUID(), DEFAULT_SERVICE, message, LogLevel.Information));
+    }
+    public static void addError(String service, String message) {
+        addLog(new Log(UUID.randomUUID(), service, message, LogLevel.Error));
+    }
+    public static void addWarning(String service, String message) {
+        addLog(new Log(UUID.randomUUID(), service, message, LogLevel.Warning));
+    }
+    public static void addInfo(String service, String message) {
+        addLog(new Log(UUID.randomUUID(), service, message, LogLevel.Information));
+    }
+    public static void addLog(String message) {
+        addLog(new Log(UUID.randomUUID(), DEFAULT_SERVICE, message, LogLevel.Information));
+    }
+    public static synchronized void addLog(Log message) {
+        System.out.println(message.service + ": " + message.message); // TODO: Remove this before merging!
```

Comment:
:round_pushpin:

Reply:
Removed in d7a829170ab7ec6a2e68d0a03988d8353afac49e